src/transformations/ForToWhileTransformation.cpp

```diff
@@ -0,0 +1,136 @@
+#include "../../include/transformations/ForToWhileTransformation.h"
+
+using clang::Expr, clang::ContinueStmt, clang::SourceRange;
+using clang::Lexer, clang::CharSourceRange;
+using llvm::isa, llvm::cast;
+using std::unique_ptr, std::find, std::vector, std::string;
+
+// ------------ ForToWhileVisitor ------------
+
+ForToWhileVisitor::ForToWhileVisitor(Rewriter * rewriter) :
+        rewriter(rewriter), sm(rewriter->getSourceMgr()), opt(rewriter->getLangOpts()) {}
+
+bool ForToWhileVisitor::VisitForStmt(ForStmt * forStmt) {
+    const Stmt * body = forStmt->getBody();
+
+    if (!body) {
+        return true;
+    }
+
+    processInit(forStmt);
+    processCond(forStmt);
+    processInc(forStmt);
+    processBody(forStmt);
+
+    return true;
+}
+
+void ForToWhileVisitor::processBody(ForStmt *forStmt) {
+    const Stmt * body = forStmt->getBody();
+    // If the inc part of for is empty, then it is not necessary to add brackets
+    if (forStmt->getInc()) {
+        if (!isa<clang::CompoundStmt>(body)) {
+            rewriter->InsertText(body->getBeginLoc(), "{\n", true, true);
+            rewriter->InsertText(body->getEndLoc().getLocWithOffset(1), "}", true, true);
+        }
+    }
+}
+
+void ForToWhileVisitor::processInit(ForStmt * forStmt) {
+    Stmt * init = forStmt->getInit();
+    if (init) {
+        SourceRange initRange = init->getSourceRange();
+        if (isa<clang::ValueStmt>(*init)) {
+            initRange.setEnd(init->getEndLoc().getLocWithOffset(1));
+        }
+        auto initText = Lexer::getSourceText(CharSourceRange::getCharRange(initRange), sm, opt).str() + ";\n";
+        rewriter->InsertText(forStmt->getBeginLoc(), initText, true, true);
+    }
+}
+
+void ForToWhileVisitor::processCond(ForStmt * forStmt) {
+    Expr * cond = forStmt->getCond();
+    string condText = "";
+    if (cond) {
+        SourceRange condRange = cond->getSourceRange();
+        if (isa<clang::ValueStmt>(*cond->getExprStmt())) {
+            condRange.setEnd(cond->getEndLoc().getLocWithOffset(1));
+        }
+        condText = Lexer::getSourceText(CharSourceRange::getCharRange(condRange), sm, opt).str();
+    } else {
+        condText = "1";
+    }
+    auto whileText = "while(" + condText + ") ";
+    ptrdiff_t _diff = sm.getCharacterData(forStmt->getBody()->getBeginLoc()) -
+                      sm.getCharacterData(forStmt->getBeginLoc());
+    auto diff = (unsigned int) _diff;
```

Comment:
Let's cast in the next line when _diff is passed to the function. Or you can cast in the previous line.